From 76b048bf6bea5db4fb40538c27146e684858da32 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:53 -0400
Subject: [tools] perf/trace: Fix summary percentage when processing files

Message-id: <1408953210-26343-106-git-send-email-jolsa@redhat.com>
Patchwork-id: 88817
O-Subject: [PATCH RHEL7.1 BZ1133083 105/322] perf trace: Fix summary percentage when processing files
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 3160565f0e005d2ec736ae25cf0a79988c0cbe71
Author: David Ahern <dsahern@gmail.com>
Date: Wed Dec 4 19:41:41 2013 -0700

description
===========
Getting a divide by 0 when events are processed from a file:

   perf trace -i perf.data -s
   ...
   dnsmasq (1684), 10 events, inf%, 0.000 msec

The problem is that the event count is not incremented as events are
processed. With this patch:

   perf trace -i perf.data -s
   ...
   dnsmasq (1684), 10 events, 8.9%, 0.000 msec
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index a7aa771..56bbca5 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -1770,8 +1770,10 @@ static int trace__process_sample(struct perf_tool *tool,
  if (!trace->full_time && trace->base_time == 0)
   trace->base_time = sample->time;
 
- if (handler)
+ if (handler) {
+  ++trace->nr_events;
   handler(trace, evsel, sample);
+ }
 
  return err;
 }
-- 
1.7.1